modular_skyrat/modules/holomap/code/holomap_datum.dm

```diff
@@ -0,0 +1,57 @@
+// Simple datum to keep track of a running holomap. Each machine capable of displaying the holomap will have one.
+/datum/station_holomap
+	var/image/base_map
+	var/image/cursor
+	var/image/legend
+
+	var/map_x
+	var/map_y
+	var/map_z
+
+/datum/station_holomap/New()
+	. = ..()
+	cursor = image('icons/holomap_markers.dmi', "you")
+	legend = image('icons/64x64.dmi', "legend")
+
+/datum/station_holomap/proc/initialize_holomap(map_x, map_y, map_z, var/mob/user = null, var/reinit_base_map = FALSE, extra_overlays = list())
+	src.map_x = map_x
+	src.map_y = map_y
+	src.map_z = map_z
+
+	if(!base_map || reinit_base_map)
+		base_map = image(SSholomaps.extra_holomaps["[HOLOMAP_EXTRA_STATIONMAP]_[map_z]"])
+
+	if(isAI(user) || isaicamera(user))
+		var/turf/eye_turf = get_turf(user.client.eye)
```

Comment:
👀 maybe its not important but client?.

Reply:
Very much so, I just haven't tested a single thing with AI yet.